armory/art_experimental/attacks/poison_loader_audio.py

```diff
@@ -60,6 +60,7 @@ def insert(self, x: np.ndarray) -> np.ndarray:
             raise ValueError("Shift + Backdoor length is greater than audio's length.")
 
         audio[shift : shift + bd_length] += self.scaled_trigger
+        audio = np.clip(audio, -1.0, 1.0)
```

Comment:
just to get more context, was the previous absence of clipping here a bug?

Reply:
David copied and modified this from ART because their version did something very slowly.  Clipping does not happen in the ART version either, but I believe it is a bug to return an audio sample that exceeds the valid range.  I filed an issue in ART for this as well.